### PR TITLE
lots of Fixes

### PR DESCRIPTION
**Pull request by stefanpenner**

### CODE REVIEW

**Comment by stefanpenner** [comment]
I didn't mean to merge this without review, so i reverted for now. @chancancode if you don't have issues i'll revert the revert and we can move on, if you have issues leave line comments here and I'll address.

File: index.js [hunk 1]
```diff
@@ -101,14 +112,5 @@ TypeScript.prototype.cacheKeyProcessString = function(string, relativePath) {
...
 
 TypeScript.prototype.processString = function (string, relativePath) {
-  try{
-    return ts.transpileModule(string, {compilerOptions: this.options, fileName: relativePath}).outputText;
-  }catch(e){
-    console.error('TYPESCRIPT ERROR:');
-    console.error(e.stack + '\n');
-
-    return '';
-  }
```
Comment (chancancode) [inline index.js:111]: Is this not necessary anymore?

Comment (stefanpenner) [inline index.js:111]: If there is a compiler error, we want that build attempt to fail and reject. This allows us to display helpful error messages to the user and perform needed cleanup.

Comment (chancancode) [inline index.js:111]: I can't remember the details, but I believe it the typescript compiler would throw errors quite liberally in cases where it doesn't matter (empty file, .d.ts or something like that). Anyhow, if you tested that it works for glimmer then it's probably fine to remove!

**Comment by chancancode** [comment]
seems good!

File: index.js (hunk 1 above)
Comment (stefanpenner) [inline index.js:111]: We can explore this further once such an example is demonstrated. Eating all errors though is likely not a good idea.

**Pull request merged**